File: apps/mobile/lib/domain/models/account.dart
/// حساب مربوط كما يراه الهاتف.
///
/// **لا اعتمادات هنا.** بعد [ADR-0009] تعيش أسرار الشركات مشفّرة على
/// السيرفر وحده؛ الهاتف يرى تسمية وحالة وعدد أجهزة، ويرسل الاعتمادات مرة
/// واحدة عند الربط ولا يحتفظ بها.
class Account {
  const Account({
    required this.id,
    required this.integrationId,
    required this.label,
    required this.status,
    this.deviceCount = 0,
    this.credentialsExpireAt,
    this.lastCheckedAt,
  });

  final String id;
  final String integrationId;
  final String label;
  final AccountStatus status;
  final int deviceCount;

  /// انتهاء اشتراك المشروع لدى الشركة — عليه يعمل تنبيه الصلاحية.
  final DateTime? credentialsExpireAt;
  final DateTime? lastCheckedAt;

  /// هل يحتاج تدخّل المستخدم؟ يُعرض له تنبيه بدل أن يكتشف العطل حين لا
  /// يستجيب جهازه.
  bool get needsAttention => status != AccountStatus.active;

  /// رسالة عربية تشرح الحالة وما العمل.
  String get statusMessage => switch (status) {
    AccountStatus.active => 'يعمل',
    AccountStatus.invalidCredentials =>
      'رفضت الشركة بيانات هذا الحساب — أعد إدخالها.',
    AccountStatus.expired =>
      'انتهى اشتراك مشروعك لدى الشركة — جدّده ثم أعد المحاولة.',
    AccountStatus.disabled => 'الحساب موقوف.',
  };

  factory Account.fromJson(Map<String, dynamic> json) => Account(
    id: json['id'] as String? ?? '',
    integrationId: json['integrationId'] as String? ?? '',
    label: json['label'] as String? ?? '',
    status: _statusFromWire(json['status'] as String?),
    deviceCount: (json['deviceCount'] as num?)?.toInt() ?? 0,
    credentialsExpireAt: DateTime.tryParse(
      json['credentialsExpireAt'] as String? ?? '',
    )?.toLocal(),
    lastCheckedAt: DateTime.tryParse(
      json['lastCheckedAt'] as String? ?? '',
    )?.toLocal(),
  );

  static AccountStatus _statusFromWire(String? wire) => switch (wire) {
    'invalid_credentials' => AccountStatus.invalidCredentials,
    'expired' => AccountStatus.expired,
    'disabled' => AccountStatus.disabled,
    _ => AccountStatus.active,
  };

  @override
  String toString() => 'Account($id, $integrationId, "$label", ${status.name})';
}

enum AccountStatus { active, invalidCredentials, expired, disabled }

File: packages/shared/src/account.ts
import { Type, type Static } from '@sinclair/typebox';

/**
 * حالة الحساب المربوط — يقرأها الويب ليعرض تنبيهاً قبل أن يفاجأ
 * المستخدم بتوقّف أجهزته (الدراسة § 7: انتهاء اشتراك Tuya).
 */
export const AccountStatus = Type.Union(
  [
    Type.Literal('active', { description: 'يعمل' }),
    Type.Literal('invalid_credentials', { description: 'اعتمادات مرفوضة — يلزم إعادة الإعداد' }),
    Type.Literal('expired', { description: 'انتهى اشتراك المشروع لدى الشركة' }),
    Type.Literal('disabled', { description: 'أوقفه المستخدم' }),
  ],
  { $id: 'AccountStatus' },
);
export type AccountStatus = Static<typeof AccountStatus>;

/**
 * حساب مربوط كما يراه العميل — **بلا أي سرّ**: الاعتمادات تُشفَّر على
 * السيرفر ولا تخرج منه أبداً (الهيكلية § 7).
 */
export const Account = Type.Object(
  {
    id: Type.String({ format: 'uuid' }),
    integrationId: Type.String({ minLength: 1 }),
    label: Type.String(),
    status: Type.Ref(AccountStatus),
    deviceCount: Type.Integer({ minimum: 0 }),
    credentialsExpireAt: Type.Optional(Type.String({ format: 'date-time' })),
    lastCheckedAt: Type.Optional(Type.String({ format: 'date-time' })),
    createdAt: Type.String({ format: 'date-time' }),
  },
  { $id: 'Account', additionalProperties: false },
);
export type Account = Static<typeof Account>;

/** قيم حقول الاعتماد بمفاتيح `IntegrationInfo.fields`. */
export const Credentials = Type.Record(Type.String(), Type.String(), {
  $id: 'Credentials',
  description: 'قيم حقول الاعتماد — تُشفَّر فور وصولها ولا تُعاد أبداً',
});
export type Credentials = Static<typeof Credentials>;

export const CreateAccountRequest = Type.Object(
  {
    integrationId: Type.String({ minLength: 1 }),
    label: Type.String({ minLength: 1, maxLength: 60 }),
    credentials: Type.Ref(Credentials),
  },
  { $id: 'CreateAccountRequest', additionalProperties: false },
);
export type CreateAccountRequest = Static<typeof CreateAccountRequest>;

/** تعديل جزئي: تسمية فقط، أو اعتمادات جديدة تُتحقّق قبل الحفظ. */
export const UpdateAccountRequest = Type.Object(
  {
    label: Type.Optional(Type.String({ minLength: 1, maxLength: 60 })),
    credentials: Type.Optional(Type.Ref(Credentials)),
  },
  { $id: 'UpdateAccountRequest', additionalProperties: false },
);
export type UpdateAccountRequest = Static<typeof UpdateAccountRequest>;

export const AccountList = Type.Object(
  { accounts: Type.Array(Type.Ref(Account)) },
  { $id: 'AccountList', additionalProperties: false },
);
export type AccountList = Static<typeof AccountList>;

/** نتيجة مزامنة حساب — كم جهازاً ظهر، وكم اختفى. */
export const SyncResult = Type.Object(
  {
    accountId: Type.String({ format: 'uuid' }),
    deviceCount: Type.Integer({ minimum: 0 }),
    added: Type.Integer({ minimum: 0 }),
    removed: Type.Integer({ minimum: 0 }),
    at: Type.String({ format: 'date-time' }),
  },
  { $id: 'SyncResult', additionalProperties: false },
);
export type SyncResult = Static<typeof SyncResult>;
